test/units/cli/test_galaxy.py

```diff
@@ -115,6 +117,32 @@ def test_display_galaxy_info(self):
         if display_result.find('\t\tgalaxy_tags:') > -1:
             self.fail('Expected galaxy_tags to be indented twice')
 
+    def test_execute_install(self):
+        # testing installing with insufficient information
+        gc = GalaxyCLI(args=["install"])
+        with patch('sys.argv', ["-c"]):
+            galaxy_parser = gc.parse()
+        self.assertRaises(AnsibleError, gc.run)
+
+        # installing role
+        gc = GalaxyCLI(args=["install"])
+        with patch('sys.argv', ["--offline", "-p", self.role_path, "-r", self.role_req]):
+            galaxy_parser = gc.parse()
+        super(GalaxyCLI, gc).run()
+        gc.api = ansible.galaxy.api.GalaxyAPI(gc.galaxy)
+        completed_task = gc.execute_install()
+
+        # testing correct installation
+        role_file = os.path.join(self.role_path, self.role_name)
+        self.assertTrue(os.path.exists(role_file))
+        self.assertTrue(completed_task == 0)
+
+        # cleaning up
+        gc.args = ["remove"]
```

Comment:
Preference would be for a unit test to be mocked sufficiently that a cleanup like this is not required.

However, in this case if that involves a large amount of fragile mocking, I wouldn't worry about it now
as having test coverage is better than not having it, even if it is a little messy.